lib/auth/cognito_stub_client.rb

```diff
@@ -64,9 +82,21 @@ def self.switch_to_stub
     return false if SelfService.service(:cognito_stub)
 
     real_client = SelfService.service(:cognito_client)
+    real_jwks = SelfService.service(:jwks)
     SelfService.register_service(name: :real_client, client: real_client)
+    SelfService.register_service(name: :real_jwks, client: real_jwks)
     SelfService.register_service(name: :cognito_client, client: stub_client)
+    load_jwks
     setup_stubs
     SelfService.register_service(name: :cognito_stub, client: true)
   end
+
+  def self.load_jwks
+    private_key = OpenSSL::PKey::RSA.generate(2048)
+    public_key = private_key.public_key
+    SelfService.register_service(name: :private_key, client: private_key)
```

Comment:
why do we register a simple value as a service? just use global variable?

Reply:
Its used else where and is subject to change if switch between stub auth and real auth

Reply:
then use a own class/method or a global variable (e.g. `$cognito_jwt_private_key` as a last resort though). The private key is not a service nor a client (`client: private_key`) ... it's just abusing the services and putting values into the client field...